pkgs/development/compilers/llvm/15/libcxx/default.nix

```diff
@@ -0,0 +1,108 @@
+{ lib, stdenv, llvm_meta
+, monorepoSrc, runCommand
+, cmake, python3, fixDarwinDylibNames, version
+, libcxxabi
+, enableShared ? !stdenv.hostPlatform.isStatic
+
+# If headersOnly is true, the resulting package would only include the headers.
+# Use this to break the circular dependency between libcxx and libcxxabi.
+#
+# Some context:
+# https://reviews.llvm.org/rG1687f2bbe2e2aaa092f942d4a97d41fad43eedfb
+, headersOnly ? false
+}:
+
+let
+  basename = "libcxx";
+in
+
+stdenv.mkDerivation rec {
+  pname = basename + lib.optionalString headersOnly "-headers";
+  inherit version;
+
+  src = runCommand "${pname}-src-${version}" {} ''
+    mkdir -p "$out"
+    cp -r ${monorepoSrc}/cmake "$out"
+    cp -r ${monorepoSrc}/${basename} "$out"
+    mkdir -p "$out/libcxxabi"
+    cp -r ${monorepoSrc}/libcxxabi/include "$out/libcxxabi"
+    mkdir -p "$out/llvm"
+    cp -r ${monorepoSrc}/llvm/cmake "$out/llvm"
+    cp -r ${monorepoSrc}/llvm/utils "$out/llvm"
+    cp -r ${monorepoSrc}/third-party "$out"
+    cp -r ${monorepoSrc}/runtimes "$out"
+  '';
+
+  sourceRoot = "${src.name}/runtimes";
+
+  outputs = [ "out" ] ++ lib.optional (!headersOnly) "dev";
+
+  prePatch = ''
+    cd ../${basename}
+    chmod -R u+w .
+  '';
+
+  patches = [
+    ./gnu-install-dirs.patch
+  ] ++ lib.optionals stdenv.hostPlatform.isMusl [
+    ../../libcxx-0001-musl-hacks.patch
+  ];
+
+  postPatch = ''
+    cd ../runtimes
+  '';
+
+  preConfigure = lib.optionalString stdenv.hostPlatform.isMusl ''
+    patchShebangs utils/cat_files.py
+  '';
+
+  nativeBuildInputs = [ cmake python3 ]
+    ++ lib.optional stdenv.isDarwin fixDarwinDylibNames;
+
+  buildInputs = lib.optionals (!headersOnly) [ libcxxabi ];
+
+  cmakeFlags = [
+    "-DLLVM_ENABLE_RUNTIMES=libcxx"
+    "-DLIBCXX_CXX_ABI=${lib.optionalString (!headersOnly) "system-"}libcxxabi"
```

Comment:
```suggestion
    "-DLIBCXX_CXX_ABI=${if headersOnly then "none" else "system-libcxxabi"}"
```

`llvmPackages_15.libcxx` is failing to build on aarch64-linux
<details>
    <summary>Log</summary>

    -- Adding Benchmark: vector_operations.bench.cpp
    -- ABI list file not generated for configuration aarch64-unknown-linux-gnu.libcxxabi.v1.stable.exceptions.nonew, `check-cxx-abilist` will not be available.
    -- Configuring done
    CMake Error at /build/libcxx-headers-src-15.0.6/libcxx/src/CMakeLists.txt:261 (add_custom_command):
      Error evaluating generator expression:

        $<TARGET_PROPERTY:libcxx-abi-shared,IMPORTED_LIBNAME>

      Target "libcxx-abi-shared" not found.


    CMake Error at /build/libcxx-headers-src-15.0.6/libcxx/src/CMakeLists.txt:261 (add_custom_command):
      Error evaluating generator expression:

        $<TARGET_PROPERTY:libcxx-abi-shared,IMPORTED_LIBNAME>

      Target "libcxx-abi-shared" not found.


    CMake Error at /build/libcxx-headers-src-15.0.6/libcxx/src/CMakeLists.txt:261 (add_custom_command):
      Error evaluating generator expression:

        $<TARGET_PROPERTY:libcxx-abi-shared,IMPORTED_LIBNAME>

      Target "libcxx-abi-shared" not found.


    -- Generating done
    CMake Warning:
      Manually-specified variables were not used by the project:

        BUILD_TESTING
        CMAKE_EXPORT_NO_PACKAGE_REGISTRY
        CMAKE_POLICY_DEFAULT_CMP0025


    CMake Generate step failed.  Build files cannot be regenerated correctly.
    error: builder for '/nix/store/2l0dl8hb022nk8a8asdrvhnc3fnivz1w-libcxx-headers-15.0.6.drv' failed with exit code 1;
           last 10 log lines:
           > -- Generating done
           > CMake Warning:
           >   Manually-specified variables were not used by the project:
           >
           >     BUILD_TESTING
           >     CMAKE_EXPORT_NO_PACKAGE_REGISTRY
           >     CMAKE_POLICY_DEFAULT_CMP0025
           >
           >
           > CMake Generate step failed.  Build files cannot be regenerated correctly.
           For full logs, run 'nix log /nix/store/2l0dl8hb022nk8a8asdrvhnc3fnivz1w-libcxx-headers-15.0.6.drv'.
    error: 1 dependencies of derivation '/nix/store/2q1nq0kb3cc1qfrjqjnxbwghr9gqq9qm-libcxxabi-15.0.6.drv' failed to build
    error: 1 dependencies of derivation '/nix/store/y2hw71gdlir4skv2rjkn3wk6vk4nr4ik-libcxx-15.0.6.drv' failed to build

</details>

Reply:
I have reapplied this patch in the rebase. It should build correctly 15.0.7